pkg/controller/core/core.go

```diff
@@ -91,6 +91,7 @@ func SetupControllers(mgr ctrl.Manager, qManager *queue.Manager, cc *cache.Cache
 		return "Workload", err
 	}
 	qManager.AddTopologyUpdateWatcher(cqRec)
+	cc.AddWorkloadUpdateWatcher(qRec)
```

Comment:
I think since we move the penalty map to queue manager, we should also move watchers